components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -201,109 +146,243 @@ open class FirefoxAccount {
     /// the caller must intercept that redirection, extract the `code` and `state` query parameters and call
     /// `completeOAuthFlow(...)` to complete the flow.
     open func beginOAuthFlow(scopes: [String], completionHandler: @escaping (URL?, Error?) -> Void) {
-        queue.async {
+        DispatchQueue.global().async {
             do {
-                let scope = scopes.joined(separator: " ")
-                let url = URL(string: String(freeingFxaString: try FirefoxAccountError.unwrap { err in
-                    fxa_begin_oauth_flow(self.raw, scope, err)
-                }))!
+                let url = try self.beginOAuthFlowSync(scopes: scopes)
                 DispatchQueue.main.async { completionHandler(url, nil) }
             } catch {
                 DispatchQueue.main.async { completionHandler(nil, error) }
             }
         }
     }
 
+    open func beginOAuthFlowSync(scopes: [String]) throws -> URL {
+        let scope = scopes.joined(separator: " ")
+        let ptr = try rustCall { err in
+            fxa_begin_oauth_flow(self.raw, scope, err)
+        }
+        return URL(string: String(freeingFxaString: ptr))!
+    }
+
+    open func beginPairingFlowSync(pairingUrl: String, scopes: [String]) throws -> URL {
```

Comment:
Future readers may be surprised to see that there is no asynchronous `beginPairingFlow` variant, because there's nothing in the file to indicate that the async variants are deprecated. Should we consider more explicitly deprecating them, or perhaps adding an async wrapper to this method for consistency?